commands/quote.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+const db = require('../services/db');
+
+module.exports = {
+  db_required: true,
+  message_regex: /^.(?:quote)(?: ([^ ]+)|$)(?: (.+)|$)$/,
+  allow: ({ isMod, isAuthenticated, isPM }) => isMod && isAuthenticated && !isPM,
+  response({ message_match: [, command, quote], author_match: [author], channel }) {
+    if (!command) {
+      command = 'get';
+    }
+    switch (command.toLowerCase()) {
+      case 'add':
+      case 'store':
+        return add(author, channel, quote);
+      case 'get':
+        return get(channel, quote);
+      case 'search':
+        return search(channel, quote);
+      case 'help':
+        return [
+          'Add \'quote\' to add a new quote',
+          'Get \'id\' to get a specific quote, or omit id for a random one',
+          'Search \'term\' to search for quote with a searchterm'
+        ];
+      default:
+        return 'Command to store/retrieve quotes. Use `.quote help` for detailed instructions.';
+    }
+  }
+};
+
+function add(author, channel, quote) {
+  if (!quote) {
+    return 'No quote to store.';
+  }
+  const addQuery = 'INSERT INTO `Quote` (`Author`, `Message`, `Chan`) VALUES (?, ?, ?);';
+  return db.conn
+    .query(addQuery, [author, quote, channel])
+    .then(created => `Storing quote '${quote}' from '${author}' on '${channel}', with id '${created.insertId}'`);
+}
+
+function get(channel, quote) {
+  if (quote) {
+    const getQuery = 'SELECT * FROM Quote WHERE Quote.ID = ?';
+    return db.conn
+      .query(getQuery, [quote]).get(0)
+      .then(result => {
+        return result ? `${result.Message} (submitted by ${result.Author})` : `No quote with ID ${quote}`;
+      });
+  }
+  const randomQuery = 'SELECT * FROM Quote WHERE Quote.Chan = ? ORDER BY RAND() LIMIT 1';
+  return db.conn
+    .query(randomQuery, [channel]).get(0)
+    .then(result => {
+      return result ? `${result.Message} (submitted by ${result.Author})` : 'No quotes currently stored';
+    });
+}
+
+function search(channel, quote) {
+  if (!quote) {
+    return 'No term to search for.';
+  }
+  const searchQuery = 'SELECT * FROM Quote WHERE Quote.Chan = ? AND Quote.Message LIKE ? ORDER BY RAND() LIMIT 1';
```

Comment:
Personally I'd expect this to return all results (or maybe ~5 at a time, prompting to continue for more results?) instead of just one, since if you're looking for something specific it'd be kind of annoying to keep having to reroll for it randomly :p

Reply:
I've changed this to do all of them currently and I'll add pagination another time, I think. If you require pagination to approve, that's fine but ... I'm not doing it right now :-P

Reply:
Would using `-all` fit for that? Like here:

https://github.com/pokemontrades/porygon-bot/blob/d4e70b9ecdcaa2ba462de0de4c5db1cc5837776c/commands/usernote.js#L104-L106

It's not full pagination I guess that one could be easily added (I could do it myself and make a PR once that one is merged)

Reply:
Yeah, if we use this elsewhere just now then it could be a solution.